### PR TITLE
export: support parallel dump on TiDB clustered index table

### PR DESCRIPTION
**Pull request by tangenta**
### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

File: v4/export/sql.go [hunk 1]
```diff
@@ -441,6 +442,62 @@ func GetTiDBDDLIDs(db *sql.DB) ([]string, error) {
...
+			log.Warn("unexpected json value type when decoding TiDB region key",
+				zap.String("key", key), zap.String("decode result", decodeResult), zap.Any("handle", v))
+		}
+		ret = append(ret, js)
+	}
+	return ret
+}
+
+func GetTiDBRegionStartKeys(db *sql.Conn, dbName, tableName string) []string {
+	query := "SELECT start_key FROM information_schema.tikv_region_status s, information_schema.tables t " +
```
Comment (lichunzhu) [inline v4/export/sql.go:484]: Have you tested this new dumpling manually? For example, primary key whose type is `int`, `varchar`, `datatime`, `date`...

File: v4/export/sql.go [hunk 2]
```diff
@@ -441,6 +442,62 @@ func GetTiDBDDLIDs(db *sql.DB) ([]string, error) {
...
+		"WHERE s.table_id = t.tidb_table_id AND t.db_name = ? AND t.table_name = ? AND is_index = 0 ORDER BY start_key;"
```
Comment (lichunzhu) [inline v4/export/sql.go:485]: It sames that `information_schema.tables` doesn't have column `db_name` but have column `table_schema`.
My TiDB version is v4.0.7.

Comment (tangenta) [reply v4/export/sql.go]: This should be `s.db_name`. Fixed.

**Review by lichunzhu — COMMENTED** [review]
1. It seems that `chunk-by-region` option can't control how much regions dumpling concurrently dump. Should we use `APPROXIMATE_KEYS` or something to control it?
2. Do we need a `chunk-by-region` option? I think we don't need that because I think this split solution is stable enough.
@kennytm WDYT?

**Comment by tangenta** [comment]
Done by https://github.com/pingcap/dumpling/pull/205.

**Pull request closed without merging**